lib/spack/spack/operating_systems/linux_distro.py

```diff
@@ -51,6 +51,11 @@ def __init__(self):
 
         if 'ubuntu' in distname:
             version = '.'.join(version[0:2])
+        # openSUSE Tumbleweed is a rolling release which can change
+        # more than once in a week, so set version to tumbleweed
+        elif 'opensuse-tumbleweed' in distname:
+            distname = 'opensuse'
+            version = 'tumbleweed'
```

Comment:
Can we keep `opensuse-tumbleweed` as a distro name and the glibc version as the "rolling" version:
```console
# ldd --version
ldd (GNU libc) 2.35
Copyright (C) 2022 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```
Ideally this should give `opensuse-tumbleweed2.35`. Thanks to @haampie and @trws for the discussion on slack and the idea.